django-verdant/rca/management/commands/archive_events.py

```diff
@@ -0,0 +1,161 @@
+import csv
+import sys
+import warnings
+from datetime import date
+from optparse import make_option
+
+from dateutil.relativedelta import relativedelta
+
+from django.core.management.base import BaseCommand
+from django.db import IntegrityError, transaction
+from rca.models import EventItem, EventIndex
+from wagtail.wagtailredirects.models import Redirect
+
+# Ignore all the deprecation warnings.
+# Legacy site, soon to be replaced.
+warnings.simplefilter("ignore")
+
+
+class Command(BaseCommand):
+    event_redirects = {}
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help="Dry run, do not unpublish pages or create redirects."
+        ),
+        make_option(
+            '--verbose',
+            action='store_true',
+            dest='verbose',
+            default=True,
+            help="Provide extended feedback."
+        ),
+        make_option(
+            '--examples',
+            action='store_true',
+            dest='examples',
+            default=False,
+            help="Provide urls for example event content."
+        ),
+        make_option(
+            '--csv',
+            action='store_true',
+            dest='output_csv',
+            default=False,
+            help="List archived event items. Negates verbose."
+        ),
+        make_option(
+            '--permanent',
+            action='store_true',
+            dest='permanent',
+            default=True,
+            help="Create permanent redirects (301)."
+        ),
+    )
+
+    def create_redirect(self, e, target_page, permanent=True):
+        new_redirect = Redirect()
+        new_redirect.site = e.get_site()
+        # Path includes 'home' and a trailing slash
+        # both must be removed.
+        bare_path = '/' + '/'.join(e.url_path.split('/')[2:][:-1])
```

Comment:
`get_url_parts ()` might make this easier: https://github.com/wagtail/wagtail/blob/master/wagtail/core/models.py#L748

Reply:
Absolutely! I have seen that before, must have drifted out of memory.